query-service-impl/src/test/java/org/hypertrace/core/query/service/trino/QueryRequestToTrinoSQLConverterTest.java

```diff
@@ -635,13 +635,13 @@ void testQueryWithContainsKeyLikeOperator() {
 
     assertSQLQuery(
         builder.build(),
-        "SELECT cast(tags as text) FROM public.\"span-event-view\" "
+        "SELECT tags FROM span-event-view "
             + "WHERE "
             + tableDefinition.getTenantIdColumn()
             + " = '"
             + TENANT_ID
             + "' "
-            + "AND tags::jsonb::text ~* '.*\"my_tag_name.*\":.*'",
+            + "AND any_match(map_keys(tags), k -> regexp_like(k, 'my_tag_name.*'))",
```

Comment:
Costly, but don't see an option here right now.

Reply:
Yes couldn't see any other alternate as per the available Trino functions.